utils/scarb-ui/src/lib.rs

```diff
@@ -119,6 +119,22 @@ impl Ui {
         self.print(TypedMessage::styled("error", "red", message.as_ref()))
     }
 
+    /// Print a warning to the user.
+    pub fn warn_with_code(&self, code: impl AsRef<str>, message: impl AsRef<str>) {
+        self.print(
+            TypedMessage::styled("warn", "yellow", message.as_ref())
+                .with_type_suffix_for_text(format!("[{}]", code.as_ref()).as_str()),
+        )
+    }
+
+    /// Print an error to the user.
+    pub fn error_with_code(&self, code: impl AsRef<str>, message: impl AsRef<str>) {
+        self.print(
+            TypedMessage::styled("error", "red", message.as_ref())
+                .with_type_suffix_for_text(format!("[{}]", code.as_ref()).as_str()),
+        )
+    }
```

Comment:
Hmm, what do you think about just having `error_code` field in the JSON output that exists if the code is known?